src/hotspot/share/utilities/debug.cpp

```diff
@@ -76,8 +76,19 @@ static intx g_asserting_thread = 0;
 static void* g_assertion_context = nullptr;
 #endif // CAN_SHOW_REGISTERS_ON_ASSERT
 
-// Set to suppress secondary error reporting.
-bool Debugging = false;
+bool DebuggingContext::_enabled = false;
+
+DebuggingContext::DebuggingContext() {
+  // Not an assert, since asserts are disabled by DebuggingContext.
+  if (is_enabled()) {
+    fatal("Multiple Debugging contexts");
+  }
```

Comment:
This seems too restrictive as you could hit different DebuggingContexts in different threads. ??

Reply:
This facility is only intended for use by manually invoked commands while the program is stopped in a debugger. Multi-threaded use is not an issue (and was not supported previously either). I don't think there are any nested uses either, but I've now run across a couple of places where nesting could be useful. So I'm changing the state from a simple bool to a nesting counter.

Reply:
Okay I hadn't realized the process was basically "suspended" when this was activated - as I said I never use this stuff.